controllers/productController.js

```diff
@@ -512,8 +521,15 @@ export const productFiltersController = async (req, res) => {
 // product count
 export const productCountController = async (req, res) => {
   try {
+    const { checked, radio } = req.query;
+
+    let args = {};
+
+    if (checked && checked.length > 0) args.category = checked;
+    if (radio && radio.length > 0)
+      args.price = { $gte: radio[0], $lte: radio[1] };
     // Opt to use countDocuments to ensure productCountController always returns correct count
-    const total = await productModel.find({}).countDocuments();
+    const total = await productModel.countDocuments({});
```

Comment:
Hmm I think this should be below:

// Applies the filters
const total = await productModel.countDocuments(args);